src/source.rs

```diff
@@ -242,17 +242,13 @@ where
         codec_override: Option<String>,
         data: Vec<u8>,
         meta: Option<StaticValue>, // See: https://github.com/rust-lang/rust/issues/63033
-    ) -> Vec<Result<LineValue>> {
+    ) -> Vec<Result<EventPayload>> {
         let mut results = vec![];
         match self.handle_pp(stream, ingest_ns, data) {
             Ok(data) => {
                 let meta_value = meta.map_or_else(Value::object, |m| m.0);
                 for d in data {
-                    let line_value = LineValue::try_new(vec![Pin::new(d)], |mutd| {
-                        // this is safe, because we get the vec we created in the previous argument and we now it has 1 element
-                        // so it will never panic.
-                        // take this, rustc!
-                        let mut_data = unsafe { mutd.get_unchecked_mut(0).as_mut().get_mut() };
+                    let line_value = EventPayload::try_new::<RentalSnot, _>(d, |mut_data| {
```

Comment:
Guessing this is part of another refactoring. Looking forward to this going away too 😄  in due course!

Reply:
Ja I wanted to keep the surface as small as possible so the only things I changed are the absolutely minimal changes. There is a lot of cleanup that is made possible by the rental removal but I went for a small reviewable change so we can separate the effects of the rental removal from the removal itself and argue about safety on a minimal changeset

Reply:
was tiny and with you both already having :+1: 'ed this I'll threw it in :D